js/classes/entities/midground/weapon/slingshot.js

```diff
@@ -52,34 +52,21 @@ class Slingshot {
         ASSET_MGR.stopAudio(SFX.SLINGSHOT_STRETCH.path);
         
         // choose from 4 different firing sounds
-        let rand = Math.floor(Math.random() * 4);
-        switch (rand) {
-            case 0:
-                ASSET_MGR.playAudio(SFX.SLINGSHOT_LAUNCH1.path, SFX.SLINGSHOT_LAUNCH1.volume);
-                break;
-            case 1:
-                ASSET_MGR.playAudio(SFX.SLINGSHOT_LAUNCH2.path, SFX.SLINGSHOT_LAUNCH2.volume);
-                break;
-            case 2:
-                ASSET_MGR.playAudio(SFX.SLINGSHOT_LAUNCH3.path, SFX.SLINGSHOT_LAUNCH3.volume);
-                break;
-            case 3:
-                ASSET_MGR.playAudio(SFX.SLINGSHOT_LAUNCH4.path, SFX.SLINGSHOT_LAUNCH4.volume);
-                break;
-        }
+        const rand = Math.floor(Math.random() * 4) + 1;
+        const sfx = SFX["SLINGSHOT_LAUNCH" + rand];
+        ASSET_MGR.playAudio(sfx.path, sfx.volume);
 
         this.isFiring = true;
         //this.startX = 26; // slingshot firing frame
 
-        INVENTORY.decreaseAmmo();
-        let ammo = INVENTORY.getCurrentAmmo();
-
-        // create a projectile and launch it in the direction of the mouse
-        GAME.addEntity(new Projectile(
-            Projectile.STONE,
-            Vector.round(this.pos),
-            Vector.round(Vector.canvasToWorldSpace(GAME.mousePos))));
-        // console.log("slingshot fired from (" + start.x + ", " + start.y + ") to (" + end.x + ", " + end.y + ")");
+        let ammoType = INVENTORY.useCurrentAmmo();
```

Comment:
I like the way you designed this method interface; the way it decreases the ammo amount and specifies the selected ammo type is convenient.